pkg/statistics/handle/autoanalyze/refresher/refresher_test.go

```diff
@@ -376,3 +377,234 @@ func TestCheckIndexesNeedAnalyze(t *testing.T) {
+		defs                       []model.PartitionDefinition
+		autoAnalyzeRatio           float64
+		currentTs                  uint64
+		wantAvgChangePercentage    float64
+		wantAvgSize                float64
+		wantAvgLastAnalyzeDuration time.Duration
+		wantPartitions             []string
+	}{
+		{
+			name: "Test Table not analyzed",
+			tblInfo: &model.TableInfo{
+				Indices: []*model.IndexInfo{
+					{
+						ID:    1,
+						Name:  model.NewCIStr("index1"),
+						State: model.StatePublic,
+					},
+				},
+				Columns: []*model.ColumnInfo{
+					{
+						ID: 1,
+					},
+					{
+						ID: 2,
+					},
+				},
+			},
+			partitionStats: map[int64]*statistics.Table{
+				1: {
+					HistColl: statistics.HistColl{
+						Pseudo:        false,
+						RealtimeCount: exec.AutoAnalyzeMinCnt + 1,
+					},
+					Version: currentTs,
+				},
+				2: {
+					HistColl: statistics.HistColl{
+						Pseudo:        false,
+						RealtimeCount: exec.AutoAnalyzeMinCnt + 1,
+					},
+					Version: currentTs,
+				},
+			},
+			defs: []model.PartitionDefinition{
+				{
+					ID:   1,
+					Name: model.NewCIStr("p0"),
+				},
+				{
+					ID:   2,
+					Name: model.NewCIStr("p1"),
+				},
+			},
+			autoAnalyzeRatio:           0.5,
+			currentTs:                  currentTs,
+			wantAvgChangePercentage:    1,
+			wantAvgSize:                2002,
+			wantAvgLastAnalyzeDuration: 0,
+			wantPartitions:             []string{"p0", "p1"},
+		},
+		{
+			name: "Test Table analyzed and only one partition meets the threshold",
+			tblInfo: &model.TableInfo{
+				Indices: []*model.IndexInfo{
+					{
+						ID:    1,
+						Name:  model.NewCIStr("index1"),
+						State: model.StatePublic,
+					},
+				},
+				Columns: []*model.ColumnInfo{
+					{
+						ID: 1,
+					},
+					{
+						ID: 2,
+					},
+				},
+			},
+			partitionStats: map[int64]*statistics.Table{
+				1: {
+					HistColl: statistics.HistColl{
+						Pseudo:        false,
+						RealtimeCount: exec.AutoAnalyzeMinCnt + 1,
+						ModifyCount:   (exec.AutoAnalyzeMinCnt + 1) * 2,
+						Columns: map[int64]*statistics.Column{
+							1: {
+								StatsVer: 2,
+							},
+							2: {
+								StatsVer: 2,
+							},
+						},
+					},
+					Version: currentTs,
+				},
+				2: {
+					HistColl: statistics.HistColl{
+						Pseudo:        false,
+						RealtimeCount: exec.AutoAnalyzeMinCnt + 1,
+						ModifyCount:   0,
+						Columns: map[int64]*statistics.Column{
+							1: {
+								StatsVer: 2,
+							},
+							2: {
+								StatsVer: 2,
+							},
+						},
+					},
+					Version: currentTs,
+				},
+			},
+			defs: []model.PartitionDefinition{
+				{
+					ID:   1,
+					Name: model.NewCIStr("p0"),
+				},
+				{
+					ID:   2,
+					Name: model.NewCIStr("p1"),
+				},
+			},
+			autoAnalyzeRatio:           0.5,
+			currentTs:                  currentTs,
+			wantAvgChangePercentage:    2,
+			wantAvgSize:                2002,
+			wantAvgLastAnalyzeDuration: 0,
```

Comment:
We should test some realistic value here. But it was blocked by https://github.com/pingcap/tidb/pull/51287.